### PR TITLE
Single build and push

### PR DESCRIPTION
**Pull request by ydye**
1: refactor build and push script.
2: enable Single image build and push.

### CODE REVIEW

**Comment by fanyangCS** [comment]
do you need to update readme?

File: service-deployment/docker_build.py [hunk 1]
```diff
@@ -283,6 +312,7 @@ def hadoop_binary_prepare(custom_hadoop_path, hadoop_version):
 def main():
     parser = argparse.ArgumentParser()
     parser.add_argument('-p', '--path', required=True, help="path to cluster configuration file")
+    parser.add_argument('-n', '--name', default='all', help="Build and push an image to the registry")
```
Comment (fanyangCS) [inline service-deployment/docker_build.py]: >('-n', '--name', d [](start = 23, length = 18)

need to update readme?

**Comment by ydye** [comment]
@fanyangCS  I will continue to develop single service deploy script. And then will update the readme together. And this PR's update is compatible with our previous build process.

File: service-deployment/docker_build.py (hunk 1 above)
Comment (ydye) [reply service-deployment/docker_build.py]: Expose it in the readme.md in the future.

Comment (fanyangCS) [inline service-deployment/docker_build.py]: >name [](start = 33, length = 4)

imagename?

Comment (ydye) [reply service-deployment/docker_build.py]: yes

**Pull request merged**